// -----------------------------------------
// 账号验证
// -----------------------------------------
syntax = "proto2";
// 玩家设备
message DPhone{
	required string deviceId    = 1;	// 设备ID
	required string deviceModel = 2;	// 设备型号（小米6，小米5）
	required string deviceSys   = 3;	// 设备系统号
	required string deviceRam   = 4;	// 设备Ram
	required string mac         = 5;	// mac地址
	optional string idfa        = 6;	// ios专用
	required int32 os           = 7;	// 设备系统（2=android，3=ios，4-pc）
	optional string language    = 8;	// 正在使用的语种
}

// 登录请求
message CSLogin {
	//option (msgid) = 101;
	required string account   		= 1;  // 用户名
	optional string password  		= 2;  // 密码
	optional string token 	  		= 3;  // token
	optional string channelId 		= 4;  // 渠道id
	optional string mediaId   		= 5;  // 子渠道id
	optional string version   		= 6;  // 客户端版本号
	optional int32 resVersion 		= 7;  // 客户端资源版本号
	optional DPhone phoneInfo 		= 8;  // 玩家设备
	optional int32 sdkType	  		= 9;  // sdk平台，平台id
	optional int32 serverUUID 		= 10; // 玩家选择的服务器入口
	optional string channelPlatform = 11; // SDK返回的渠道名称
}

// 登录请求返回结果
message SCLoginResult {
	//option (msgid) = 102;
	optional int32 resultCode   		= 1;	// 登录结果码
	optional string account     		= 2;	// 真实账号，登录成功才有
	optional int32 zoneTime     		= 3;	// 用服务器本地时区转出2010年1月1日 零点
	optional int32 serverId	  			= 4;	// 当前服务器id（即服务器所属的envid）
	optional string gameAddress 		= 5;	// 分配的游戏服务器的地址，ip:port
	optional int64 sessionId 	  		= 6;	// 会话id，服务器的连接id
	optional int32 timeout 	  			= 7;	// 客户端如果在timeout秒内收不到CSPing的回复就会主动断开
	optional bool prepareReconnection	= 8;	// 是否准备重连，只有在登录时发现该账号还没登出才有效
	optional bool isNeedUpdate 			= 9; 	// 是否需要更新
	optional string failText     		= 10;	// 失败时的文本内容
}

// 登陆到游戏服
message CSLoginGate {
	//option (msgid) = 103;
	required string account   		= 1;	// 账号
	required int64 sessionId  		= 2;	// 会话id
	optional string token 	  		= 3;	// token
	optional DPhone phoneInfo 		= 4;	// 玩家设备
	optional int32 sdkType	  		= 5;	// sdk平台，平台id
	optional int32 serverUUID 		= 6;	// 玩家选择的服务器入口
	optional string channelPlatform = 7; 	// SDK返回的渠道名称
}

// 登陆到游戏服返回
message SCLoginGateResult {
	//option (msgid) = 104;
	required int32 errorCode = 1;	// 登录gate错误码
}

// 请求重连
message CSAccountReconnect {
	//option (msgid) = 105;
	required int64 humanId    = 1;	// 玩家id
	required int64 sessionId  = 2;	// 会话id
	optional string token     = 3;	// token
	optional string account   = 4;	// 账号
}

// 请求重连返回结果
message SCAccountReconnectResult {
	//option (msgid) = 106;
	optional int32 errorCode = 1; // 错误编号
}

// 请求一个名字
message CSAccountRandomName {
	//option (msgid) = 107;
	optional int32 sex = 1; // 性别
}

// 请求一个名字的返回
message SCAccountRandomNameResult {
	//option (msgid) = 108;
	optional string name = 1;	// 名字
}

// 登陆的排队状态
message SCAccountLoginQueue {
	//option (msgid) = 109;
	optional int32 num = 1;	// 排队人数
	optional int32 sec = 2;	// 排队时间
	optional bool full = 3;	// 服务器已满
}

// 封号信息
message SCAccountLoginForbid {
	//option (msgid) = 122;
	optional string tip = 1;	// 提示
	optional int64 milliSec = 2;	// 剩余毫秒数
}

// 禁言信息
message SCAccountTalkForbid {
	//option (msgid) = 124;
	optional string tip = 1;	// 提示
	optional int64 milliSec = 2;	// 剩余毫秒数
}